src/overlays/actors/ovl_Bg_Astr_Bombwall/z_bg_astr_bombwall.c

```diff
@@ -28,59 +37,191 @@ const ActorInit Bg_Astr_Bombwall_InitVars = {
+void func_80C09ED0(ColliderTrisInit* arg0, Vec3f* arg1, Vec3s* arg2, ColliderTris* arg3) {
+    s32 i;
+    s32 j;
+    Vec3f sp54[3];
+
+    Matrix_RotateYS(arg2->y, MTXMODE_NEW);
+    Matrix_RotateXS(arg2->x, MTXMODE_APPLY);
+    Matrix_RotateZS(arg2->z, MTXMODE_APPLY);
+
+    for (i = 0; i < arg0->count; i++) {
+        for (j = 0; j < 3; j++) {                                          // https://decomp.me/scratch/JrEnl
+            Matrix_MultVec3f(&(arg0->elements + i)->dim.vtx[j], &sp54[j]); //! FAKE MATCH:
+            Math_Vec3f_Sum(&sp54[j], arg1, &sp54[j]);
+        }
+        Collider_SetTrisVertices(arg3, i, &sp54[0], &sp54[1], &sp54[2]);
+    }
+}
+
+void BgAstrBombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgAstrBombwall* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_astr_obj_Colheader_002498);
+    Collider_InitTris(globalCtx, &this->collider);
+    if (Flags_GetSwitch(globalCtx, OBJBgAstrBombwall_GET_SWITCHFLAG(this))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    this->dyna.actor.flags |= 0x10000000;
+    if (Collider_SetTris(globalCtx, &this->collider, &this->dyna.actor, &sTrisInit, this->colliderElements) == 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    func_80C09ED0(&sTrisInit, &this->dyna.actor.world.pos, &this->dyna.actor.shape.rot, &this->collider);
+    SubS_FillCutscenesList(&this->dyna.actor, this->unk238, ARRAY_COUNT(this->unk238));
+    func_80C0A378(this);
+}
+
+void BgAstrBombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgAstrBombwall* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80C0A120(BgAstrBombwall* this, GlobalContext* globalCtx) {
+    s32 var_s1;
+    Vec3f spC8;
+    Vec3f spBC;
+    s16 var_v0;
+    Vec3f spAC;
+    f32 temp_fv0;
+
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    for (var_s1 = 0; var_s1 < 0x1E; var_s1++) {
+        spC8.x = Rand_Centered() * 140.0f;
+        spC8.y = Rand_ZeroOne() * 200.0f;
+        spC8.z = 0.0f;
+        Matrix_MultVec3f(&spC8, &spBC);
+        Math_Vec3f_Sum(&this->dyna.actor.world.pos, &spBC, &spBC);
+        func_800BBFB0(globalCtx, &spBC, 50.0f, 2, Rand_ZeroOne() * 120.0f + 20.0f, Rand_ZeroOne() * 240.0f + 20.0f, 0);
+        spAC.x = Rand_ZeroOne() * 2.5f;
+        spAC.y = (Rand_ZeroOne() * 2.5f) + 1.0f;
+        spAC.z = Rand_ZeroOne() * 2.5f;
+        temp_fv0 = Rand_ZeroOne();
+
+        if (temp_fv0 < 0.2f) {
+            var_v0 = 0x60;
+        } else if (temp_fv0 < 0.6f) {
+            var_v0 = 0x40;
+        } else {
+            var_v0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &spBC, &spAC, &spBC, -0x104, var_v0, 0x14, 0, 0, 0xA, 0, 0, 0x32, -1, 0x267,
+                             object_astr_obj_DL_002178);
```

Comment:
```suggestion
        EffectSsKakera_Spawn(globalCtx, &spBC, &spAC, &spBC, -260, var_v0, 20, 0, 0, 10, 0, 0, 0x32, -1, OBJECT_ASTR_OBJ,
                             object_astr_obj_DL_002178);
```
From the function declaration in functions.h as well as from other uses of the function we see most of these args are used as decimal, and that the second to last argument is an objId so should use the Object enum in z64object.h.